src/logging.rs

```diff
@@ -342,6 +364,24 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_priority_as_str_matches_to_string() {
+        let priorities = [
+            Priority::Emergency,
+            Priority::Alert,
+            Priority::Critical,
+            Priority::Error,
+            Priority::Warning,
+            Priority::Notice,
+            Priority::Info,
+            Priority::Debug,
+        ];
+
+        for priority in &priorities {
+            assert_eq!(&(u8::from(priority.clone())).to_string(), priority.as_str());
```

Comment:
`priority` is `Copy`; we do you need to `clone` here?

Also, this test seems to be some kind of circular reasoning, since the `From<u8>` instance is also our own code.

Reply:
`priority` is a `&Priority`, the way that trait implementations end up working, I get an error if I don't explicitly provide a value that is a `Priority`.  The other alternative was dereferencing the reference and forcing a copy, but that seemed less clear.  I've changed it to that (i.e. `*priority`) now.

The test is less circular and just enforcing the the internal function matches what's available via the public API + `to_string()`.

Reply:
Okay, if we read this test as checking that an internal representation matches the public API the test provides value 👍

Concerning the priority reference, does it have to be a reference?  Can't we iterate over the array directly instead of taking a slice?  I dimly remember there was something with editions and IntoIterator for arrays, but you should always be able to use an explicit into_iter call 🤔

I'd find it clearer to "consume" the array in this test and avoid cloning or dereferencing in the assert, to declutter the relevant comparison 🙂

But that's really just nitpicking 😇😊🙈

Reply:
I didn't think too hard about it, since it's a test, but into_iter works and simplifies things.  Thanks.

Reply:
👍🏿